### PR TITLE
Default to `-defaultlib=` (no druntime/Phobos) for *.c tests

### PR DESCRIPTION
**Pull request by kinke**

### CODE REVIEW

File: test/tools/d_do_test.d [hunk 1]
```diff
@@ -688,6 +688,10 @@ bool gatherTestParameters(ref TestArgs testArgs, string input_dir, string input_
     if (testArgs.mode == TestMode.FAIL_COMPILE)
         testArgs.requiredArgs = "-verrors=0 " ~ testArgs.requiredArgs;
 
+    // *.c tests: don't link druntime/Phobos
+    if (input_file.extension() == ".c")
```
Comment (thewilsonator) [inline test/tools/d_do_test.d:692]: `else if`

**Review by thewilsonator — APPROVED** [review]
Otherwise looks good.

**Comment by kinke** [comment]
DMD codegen doesn't look too good (this works with LDC; some importC tests are disabled though). `__builtin_va_end` is apparently no builtin but a druntime import, and TypeInfos seem to be emitted into the supposedly C object files too.

**Comment by maxhaton** [comment]
ImportC.h is in druntime and will presumably need to be tested at some point when preprocessing is more final.

**Comment by kinke** [comment]
Filed https://issues.dlang.org/show_bug.cgi?id=23078.

**Comment by thewilsonator** [comment]
https://github.com/dlang/dmd/pull/16818

**Pull request closed without merging**